aip/general/0136.md

```diff
@@ -44,25 +44,42 @@ services. The bullets below apply in all three cases.
   - The name **must not** contain prepositions ("for", "with", etc.).
   - The verb in the name **should not** contain any of the standard method verbs ([Get][],
     [List][], [Create][], [Update][], [Delete][]).
-- The HTTP method for custom methods **should** usually be `POST`, unless the
-  custom method maps more strongly to another HTTP verb.
-  - Custom methods that serve as an alternative to get or list methods (such as
-    `Search`) **should** use `GET`. These methods **must** be idempotent and
-    have no side effects.
-  - Custom methods **should not** use `PATCH` or `DELETE`.
+- The HTTP method **must** be `GET` or `POST`:
+  - `GET` **must** be used for methods retrieving data or resource state.
+  - `POST` **must** be used if the method has side effects or mutates resources
+    or data.
 - The HTTP URI **must** use a `:` character followed by the custom verb
   (`:archive` in the above example), and the verb in the URI **must** match the
   verb in the name of the RPC.
   - If word separation is required, `camelCase` **must** be used.
 - The `body` clause in the `google.api.http` annotation **should** be `"*"`.
-  - However, if using `GET` or `DELETE`, the `body` clause **must** be absent.
```

Comment:
So why don't we allow `DELETE` anymore? Or is that written elsewhere I just haven't seen?

Reply:
That guidance used to be in there:

```
  custom method maps more strongly to another HTTP verb.
  - Custom methods that serve as an alternative to get or list methods (such as
    `Search`) **should** use `GET`. These methods **must** be idempotent and
    have no side effects.
  - Custom methods **should not** use `PATCH` or `DELETE`.
```

But I removed it because "must use GET and POST" was more succinct. And I doubt we want people to add additional custom verbs.

As to why we don't want to use delete? we could, but it just makes it another rarely used category that clients have to handle. Potentially we could add it back, but I was erring toward being conservative and relaxing guidance as needed.

Reply:
Oh you are right. Thanks!